windows-gaming-driver/src/setup/usb.rs

```diff
@@ -82,18 +84,116 @@ pub fn select(machine: &mut MachineConfig) -> bool {
     true
 }
 
+/// Lets the user remove (some) of given devices and returns a tuple of the mouse and keyboard still in the list afterwards
+///
+/// This modifies the passed list.
+fn remove(usb_devices: &mut Vec<UsbBinding>) -> (Option<UsbBinding>, Option<UsbBinding>) {
+    let devs = list_devices(None).expect("Can't read connected USB devices");
+
+    loop {
+        println!("You have currently selected the following usb devices: ");
+        let mut last = 0;
+        for (i, binding) in usb_devices.iter().cloned().enumerate() {
+            if let Some(pos) = devs.iter().position(|dev| UsbBinding::from(dev.id) == binding
```

Comment:
So seeing those usages, instead of the questionable `From` impls you should probably just implement `PartialEq`.